src/libsync/propagatedownload.cpp

```diff
@@ -484,7 +486,22 @@ void PropagateDownloadFileQNAM::slotGetFinished()
         return;
     }
 
-    downloadFinished();
+    // do whatever is needed to add a checksum to the http upload request.
+    // in any case, the validator will emit signal startUpload to let the flow
+    // continue in slotStartUpload here.
+    _validator = new TransmissionChecksumValidator( _tmpFile.fileName() );
```

Comment:
You should either use a QScopedPoointer for it, or set a parent. Otherwise it might leak if the sync engine is destroyed during the sync.